tools/ds-identify

```diff
@@ -1137,8 +1145,73 @@ dscheck_SmartOS() {
     return ${DS_NOT_FOUND}
 }
 
+none_read_always_found_setting() {
+    # the 'always_found' setting for fallback datasource controls
+    # behavior of ds-identify with respect to cloud-init, whether it
+    # should be enabled or not, and wheather ds-identify should
+    # filter-out datasources to leave just the fallback datasource
+    # enabled.
+    # order of precedence is:
+    #  1. builtin setting here cloud-init/ds-identify builtin
+    #  2. ds-identify config
+    #  3. system config (/etc/cloud/cloud.cfg.d/*None*.cfg)
+    #  4. kernel command line (undocumented)
+    #  5. user-data or vendor-data (not available here)
+    local default="$1" key="ci.datasource.none.always_found" val=""
+
+    # 4. kernel command line
+    case " ${DI_KERNEL_CMDLINE} " in
+        *\ $key=*\ )
+            val=${DI_KERNEL_CMDLINE##*$key=}
+            val=${val%% *};
+            _RET=${val:-$default}
+            return 0
+    esac
+
+    # 3. look for the key 'always_found' (datasource/None/always_found)
+    # only in cloud.cfg or cloud.cfg.d/None.cfg (case insensitive)
+    local cfg="${PATH_ETC_CI_CFG}" cfg_d="${PATH_ETC_CI_CFG_D}"
+    if check_config always_found "$cfg" "$cfg_d/*[Nn][Oo][Nn][Ee]*.cfg"; then
+        debug 2 "${_RET_fname} set always_found to $_RET"
+        return 0
+    fi
+
+    # 2. ds-identify config (datasource.none.always_found)
+    local config="${PATH_DI_CONFIG}"
+    if [ -f "$config" ]; then
+        if _read_config "$key" < "$config"; then
+            _RET=${_RET:-$default}
+            return 0
+        fi
+    fi
+
+    # 1. Default
+    _RET=$default
+    return 0
+}
+
 dscheck_None() {
-    return ${DS_NOT_FOUND}
+    local default="${DI_NONE_ALWAYS_FOUND_DEFAULT}"
+    if none_read_always_found_setting "$default"; then
+        alwaysfound="$_RET"
+    else
+        debug 1 "ec2_read_strict returned non-zero: $?. using '$default'."
+        alwaysfound="$default"
+    fi
+
+    local key="datasource/none/always_found"
+    case "$alwaysfound" in
+        true|false) :;;
```

Comment:
Do we normalize the values?  what about True|False|1|0 ?

Reply:
Ec2.strict_id values are not normalized, with only strict values accepted and all others rejected.

I went with the same strict parsing of only `true` and `false` as valid values, with anything else rejected. In the `strict_id` case warn numbers have some defined meaning.